app/plugin.go

```diff
@@ -400,6 +405,61 @@ func (a *App) GetPlugins() (*model.PluginsResponse, *model.AppError) {
 	return resp, nil
 }
 
+// GetPluginPublicKeys returns all public keys listed in the config.
+func (a *App) GetPluginPublicKeys() ([]string, *model.AppError) {
+	config := a.Config().PluginSettings
+
+	return config.SignaturePublicKeyFiles, nil
+}
+
+// GetPublicKey will return the actual public key saved in the `filename` file.
+func (a *App) GetPublicKey(filename string) ([]byte, *model.AppError) {
+	if !strings.HasSuffix(filename, PluginSignaturePublicKeyFileExtention) {
+		return nil, model.NewAppError("GetPublicKey", "api.plugin.get_public_key.not_a_public_key.app_error", nil, "", http.StatusInternalServerError)
+	}
+	data, err := a.Srv.configStore.GetFile(filename)
+	if err != nil {
+		return nil, model.NewAppError("GetPublicKey", "api.plugin.get_public_key.get_file.app_error", nil, err.Error(), http.StatusInternalServerError)
+	}
+	return data, nil
+}
+
+// AddPublicKey method will add plugin public key to the config.
+func (a *App) AddPublicKey(file string) *model.AppError {
```

Comment:
`AddPublicKey(filename string, reader io.Reader) *model.AppError`

Reply:
Spoke to @iomodo. Will not be enforcing any extension for this api, `.plugin.asc` will purely be an internal detail, which will not be visible to the client of this API. When querying for public keys, `.plugin.asc` will also be hidden.